modules/http/http-grammar.ym

```diff
@@ -103,7 +104,7 @@ http_option
     | KW_USER       '(' string ')'            { http_dd_set_user(last_driver, $3); free($3); }
     | KW_PASSWORD   '(' string ')'            { http_dd_set_password(last_driver, $3); free($3); }
     | KW_USER_AGENT '(' string ')'            { http_dd_set_user_agent(last_driver, $3); free($3); }
-    | KW_HEADERS    '(' string_list ')'       { http_dd_set_headers(last_driver, $3); g_list_free($3); }
+    | KW_HEADERS    '(' string_list ')'       { http_dd_set_headers(last_driver, $3); g_list_free_full($3, g_free); }
     | KW_METHOD     '(' string ')'            { http_dd_set_method(last_driver, $3); free($3); }
```

Comment:
unfortunately the lexer does not use g_strdup() but rather strdup() (e.g. without the g_) to allocate strings, and on win32 g_strdup and strdup are not the same :(

so, please either use free() as the destroy notify callback, or fix the lexer. I've always used the first solution in the past, but it's getting really annoying.